sys/net/gnrc/network_layer/ipv6/gnrc_ipv6.c

```diff
@@ -124,19 +119,13 @@ void gnrc_ipv6_demux(kernel_pid_t iface, gnrc_pktsnip_t *pkt, uint8_t nh)
     }
 
     DEBUG("ipv6: forward nh = %" PRIu8 " to other threads\n", nh);
-    receiver_num = gnrc_netreg_num(pkt->type, GNRC_NETREG_DEMUX_CTX_ALL) +
-                   gnrc_netreg_num(GNRC_NETTYPE_IPV6, nh);
 
-    if (receiver_num == 0) {
+    if ((gnrc_netapi_dispatch_receive(pkt->type, GNRC_NETREG_DEMUX_CTX_ALL, pkt) == 0)
```

Comment:
Because, this calls `gnrc_pktbuf_hold(gnrc_netreg_num(pkt->type, GNRC_NETREG_DEMUX_CTX_ALL) - 1)`